### PR TITLE
[artf308168] : Update jenkins plugin to support both WEBR and EventQ

### PR DESCRIPTION
**Pull request by hariharan-cn**
1. Added plugin to support - post data to WEBR
2. Added Notifier and Event listener to send data
3. Modified publisher to support Jenkins pipeline projects
   - included webhook url field and removed required mandatory for EventQ serverUrl and sak
4. Modified Traceability to accept data from TF/WEBR

File changes,

1. BuildNotifier and BuildNotifierDescriptor
   - Event listener for Jenkins Run/Abstract build actions
2. Build Event
   - used to constuct the Build object from Jenkins build
3. PushNotification
   - send/publish data to WEBR rest-end point [webhook url]
4. TraceabiltyAction & TraceabilityActionFactory
   - to project traceability view on Jenkins

Resources changes,
1. BuildNotifier - config.jelly to accept TF Url, username and Webhook url
2. Traceability - included angular-cookies,md5, ui-bootstrap min.js
3. PublishEventStep - modified config.jelly to accept Webkhook url

Scripts
1. Added below js and css under scripts/webhook to support WEBR/Webhook integrations
   tf-traceability.min, angular_cookies.min, md5.min, ui-bootstrap-tpls.min, tf.css and config.css

Tests:
1. Added TestPushNotification to test notifier changes
2. Modified TestBuildNotifier and EntryPages

NOTE: Intentionally left duplicate codes at few place since we planned to remove EventQ support. Duplicates will be removed long with EventQ.

### CODE REVIEW

**Comment by hariharan-cn** [comment]
We have to introduce new class and jelly for webhook step support. I felt which is not required since we planned to remove the EventQ.

File: src/main/java/hudson/plugins/collabnet/util/Helper.java [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+            params.add(new BasicNameValuePair("scope", "urn:ctf:services:ctf " +
+                    "urn:ctf:services:svn urn:ctf:services:gerrit"));
+            post.setEntity(new UrlEncodedFormEntity(params));
+            response = httpClient.execute(post);
+            token = JSONObject.fromObject(EntityUtils.toString(response.getEntity())).
+                    get("access_token").toString();
+        } catch (Exception e) {
+            e.printStackTrace();
+        } finally {
+            response.close();
```
Comment (BoraAksoy) [inline src/main/java/hudson/plugins/collabnet/util/Helper.java]: response (and httpClient?) can be null

File: src/main/resources/jenkins/plugins/collabnet/steps/PublishWebhookStep/help-excludeCommitInfo.html [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+  ~
+  ~ Unless required by applicable law or agreed to in writing, software
+  ~ distributed under the License is distributed on an "AS IS" BASIS,
+  ~ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  ~ See the License for the specific language governing permissions and
+  ~ limitations under the License.
+  -->
+
+<div>
+    Whether to exclude associated commit information in EventQ notification.
```
Comment (BoraAksoy) [inline src/main/resources/jenkins/plugins/collabnet/steps/PublishWebhookStep/help-excludeCommitInfo.html]: Needs to be updated for CTF webhook

File: src/main/resources/jenkins/plugins/collabnet/steps/PublishWebhookStep/help-markUnstable.html [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+  ~
+  ~ Unless required by applicable law or agreed to in writing, software
+  ~ distributed under the License is distributed on an "AS IS" BASIS,
+  ~ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  ~ See the License for the specific language governing permissions and
+  ~ limitations under the License.
+  -->
+
+<div>
+    Mark current job as <strong>unstable</strong> if this step fails to notify EventQ.
```
Comment (BoraAksoy) [inline src/main/resources/jenkins/plugins/collabnet/steps/PublishWebhookStep/help-markUnstable.html]: Needs to be updated for CTF webhook

File: src/main/resources/jenkins/plugins/collabnet/steps/PublishWebhookStep/help-status.html [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+  ~
+  ~ Unless required by applicable law or agreed to in writing, software
+  ~ distributed under the License is distributed on an "AS IS" BASIS,
+  ~ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  ~ See the License for the specific language governing permissions and
+  ~ limitations under the License.
+  -->
+
+<div>
+    (Optional) Specify EventQ status explicitly. If not provided, it will be determined based on the current run result.
```
Comment (BoraAksoy) [inline src/main/resources/jenkins/plugins/collabnet/steps/PublishWebhookStep/help-status.html]: Needs to be updated for CTF webhook

**Comment by hariharan-cn** [comment]
Fixed those minor issues. I have provided the local Jenkins plugin jar/hpi to RAM to qualify this changes.

**Pull request merged**